R/endpoints.R

```diff
@@ -119,6 +119,39 @@ model_status <- function(queue) {
   }
 }
 
+model_result <- function(queue) {
+  function(id) {
+    task_status <- queue$queue$task_status(id)
+    if (task_status == "COMPLETE") {
+      hintr:::process_result(queue$result(id))
+    } else if (task_status == "ERROR") {
+      result <- queue$result(id)
+      trace <- c(sprintf("# %s", id), result$trace)
+      error_data <- structure(result$message, trace = trace)
+      pkgapi::pkgapi_stop(error_data, "MODEL_RUN_FAILED")
+    } else if (task_status == "ORPHAN") {
+      pkgapi::pkgapi_stop(tr_("MODEL_RESULT_CRASH"), "MODEL_RUN_FAILED")
```

Comment:
I don't think we should be translating the keys here: `MODEL_RESULT_CRASH` should be constant surely?

Reply:
`MODEL_RESULT_CRASH` is the key we are using for the message string which gets translated. `MODEL_RUN_FAILED` is the error key.

Reply:
oh wow, that's quite confusing to read. We should probably think about this later. Something like prefix all translate strings with "TR_"?